### PR TITLE
chore: CON-1302 Rename ECDSA metrics

### PR DESCRIPTION
**Pull request by eichhorl**
These metrics are no longer used for tECDSA exclusively, but also for tSchnorr.

The corresponding MR updating affected dashboards, runbooks and alerts can be found [here](https://gitlab.com/dfinity-lab/private/k8s/k8s/-/merge_requests/3860).

Additionally, we remove the persisted metric `replicated_state_ecdsa_signature_agreements_total`, as it is superseded by the metric vector `replicated_state_threshold_signature_agreements_total`, which counts signature agreements by `key_id`.

### CODE REVIEW

**Review by basvandijk — APPROVED** [review]

**Comment by kpop-dfinity** [comment]
> The corresponding MR updating affected dashboards, runbooks and alerts can be found [here](https://gitlab.com/dfinity-lab/private/k8s/k8s/-/merge_requests/3860).

In which order are you planning to merge the PRs/MRs?

If I understand correctly, if we change the dashboards now, they (the dashboards) will show no data until the changes from this PR are deployed to signing subnets. 

On the other hand, if we deploy the changes in this PR first, before the changes in the linked MR, then we might not get an alert if something happens in the time window between the two deployments.

Would it make sense to split the linked MR into two: one with changes to the alerts & runbooks and another with the changes to the dashboards? And then deploy them in the following order: MR with alerts & runbooks -> this PR -> MR with dashboards

What do you think?

**Comment by eichhorl** [comment]
> In which order are you planning to merge the PRs/MRs?

I plan to merge the MR making changes to the dashboards/alerts at the latest as soon as the replica version containing the changes is elected

> If I understand correctly, if we change the dashboards now, they (the dashboards) will show no data until the changes from this PR are deployed to signing subnets.

Unless we temporarily change all panels to look at both metrics (new and old), this will always happen, as throughout the week there will be times where not all 4 subnets will be running the same replica version. But I'm not sure that's worth the effort. Instead I have been keeping up a [temporary copy](https://grafana.mainnet.dfinity.network/d/Q5hlsbXSz/threshold-signatures-temp?orgId=1&refresh=10s) of the threshold signatures dashboard, that still uses the old metrics. So if at any point in time the new dashboard won't show the data, it should still be there in the old one.

**Review by rumenov — APPROVED** [review]

File: rs/protobuf/def/state/metadata/v1/metadata.proto [hunk 1]
```diff
@@ -245,7 +245,8 @@ message SubnetMetrics {
   types.v1.NominalCycles consumed_cycles_by_deleted_canisters = 1;
   types.v1.NominalCycles consumed_cycles_http_outcalls = 2;
   types.v1.NominalCycles consumed_cycles_ecdsa_outcalls = 3;
-  optional uint64 ecdsa_signature_agreements = 4;
```
Comment (schneiderstefan) [inline rs/protobuf/def/state/metadata/v1/metadata.proto:248]: Just to verify, in the currently rolled out version, we never rely on that value anywhere? So if we need to downgrade to that version, this field being set to 0 will not cause any problems?

Comment (eichhorl) [reply rs/protobuf/def/state/metadata/v1/metadata.proto:248]: Yes, that's fine. The value of `ecdsa_signature_agreements` has already been migrated to the new field `threshold_signature_agreements` on mainnet. See [here](https://victoria.ch1-obs1.dfinity.network/select/0/vmui/#/?g0.expr=max+by+%28ic_subnet%29+%28replicated_state_ecdsa_signature_agreements_total%7Bic_subnet%3D%22pzp6e-ekpqk-3c5x7-2h6so-njoeq-mt45d-h3h6c-q3mxf-vpeq5-fk5o7-yae%22%7D%29&g0.range_input=4h39m38s322ms&g0.end_input=2024-07-18T16%3A23%3A01&g0.relative_time=none&g0.tenantID=0&g1.expr=max+by+%28ic_subnet%29+%28replicated_state_threshold_signature_agreements_total%7Bic_subnet%3D%22pzp6e-ekpqk-3c5x7-2h6so-njoeq-mt45d-h3h6c-q3mxf-vpeq5-fk5o7-yae%22%7D%29&g1.range_input=4h39m38s322ms&g1.end_input=2024-07-18T16%3A23%3A01&g1.relative_time=none&g1.tenantID=0) for example.

**Review by schneiderstefan — APPROVED** [review]

**Review by dsarlis — APPROVED** [review]

**Pull request merged**